### DISCUSSION TITLE
Backup of k8s cluster along with Pv’s when cluster are running on-premise.

### DISCUSSION THREAD

**Discussion by pr07pr07** [vmware-tanzu/velero#7333] [Community support Q&A]
Hi,

My clusters currently run in an on-premises environment, utilizing an NFS server on a separate VM for storage. The connection to this storage is facilitated through an NFS client external provider installed on the cluster, enabling dynamic provisioning of Persistent Volumes (PVs).

Now, I'm exploring the possibility of taking backups of the clusters, including PVs, using Azure plugin. I'd like to know if Azure plugin supports the backup of on-premises clusters along with their PVs and Kubernetes resources. Your assistance in clarifying this matter would be greatly appreciated.

//Prathap

**Comment by draghuram** [comment]
Just to be clear, you want to backup on prem cluster resources and PV data to Azure storage? And yes, you can backup NFS PVs using file system backups (https://velero.io/docs/v1.12/file-system-backup/). Note that NFS doesn't support CSI snapshots as far as I am aware (though some provisioners may do) so file system backups may be your best bet. 

Hope this helps,
Raghu (CloudCasa).
